dbt_airflow_factory/cosmos/project_config_builder.py

```diff
@@ -0,0 +1,127 @@
+"""Builder for Cosmos ProjectConfig from dbt.yml configuration."""
+
+import json
+import os
+from typing import Any, Dict, Optional
+
+from cosmos.config import ProjectConfig
+
+
+def extract_project_name_from_manifest(manifest_path: str) -> Optional[str]:
+    """
+    Extract dbt project name from manifest.json.
+
+    The project name is embedded in every node's unique ID with the format:
+    node_type.project_name.node_name (e.g., "model.my_project.customers")
+
+    Args:
+        manifest_path: Path to manifest.json file
+
+    Returns:
+        Project name extracted from first node, or None if cannot extract
+    """
+    try:
+        if not os.path.exists(manifest_path):
+            return None
+
+        with open(manifest_path, "r") as f:
+            manifest = json.load(f)
+
+        # Try nodes first (models, tests, snapshots, etc.)
+        nodes = manifest.get("nodes", {})
+        if nodes:
+            first_node_id = next(iter(nodes.keys()))
+            parts = first_node_id.split(".")
+            if len(parts) >= 2:
+                return parts[1]  # project_name is second part
+
+        # Fallback: try sources
+        sources = manifest.get("sources", {})
+        if sources:
+            first_source_id = next(iter(sources.keys()))
+            parts = first_source_id.split(".")
+            if len(parts) >= 2:
+                return parts[1]
+
+        return None
+
+    except (json.JSONDecodeError, OSError, KeyError, StopIteration):
+        return None
+
+
+def build_project_config(
```

Comment:
Do we need to extract project name from manifest? If we were to force the name to be the dag ID, we could get rid of most of the logic here